edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/SubscriptionProcessor.cs

```diff
@@ -44,11 +45,34 @@ public SubscriptionProcessor(IConnectionManager connectionManager, IInvokeMethod
             Preconditions.CheckNotNull(deviceConnectivityManager, nameof(deviceConnectivityManager));
             this.invokeMethodHandler = Preconditions.CheckNotNull(invokeMethodHandler, nameof(invokeMethodHandler));
             this.pendingSubscriptions = new ConcurrentDictionary<string, ConcurrentQueue<(DeviceSubscription, bool)>>();
+            this.subscriptionsBeingProcessed = new ConcurrentDictionary<string, bool>();
             this.processSubscriptionsBlock = new ActionBlock<string>(this.ProcessPendingSubscriptions);
+            connectionManager.DeviceConnected += this.DeviceConnected;
             deviceConnectivityManager.DeviceConnected += this.CloudConnectivityEstablished;
             connectionManager.CloudConnectionEstablished += this.ClientCloudConnectionEstablished;
         }
 
+        async void DeviceConnected(object sender, IIdentity identity)
```

Comment:
Instead of this, can we use the same callback for both DeviceConnected and CloudConnectionEstablished and if an identity is being processed, then skip processing it again?